### PR TITLE
only report reverted tx if they are originated from our aggKey

### PR DESCRIPTION
**Pull request by marcellorigotti**

### CODE REVIEW

File: src/metrics/sol/startSubscription.ts [hunk 1]
```diff
@@ -35,12 +35,19 @@ export const startSubscription = async (context: Context) => {
...
             new PublicKey(global.solanaCurrentOnChainKey),
-            (log: any) => {
+            async (log: any) => {
                 if (log.err !== null) {
-                    metricSolanaTxReverted.labels(log.signature).set(1);
-                    setTimeout(() => {
-                        metricSolanaTxReverted.remove(log.signature);
-                    }, 600000); // 10m
+                    const transaction = await context.connection.getTransaction(log.signature, {
+                        commitment: 'finalized',
```
Comment (albert-llimos) [inline src/metrics/sol/startSubscription.ts:41]: the subscription doesn't seem to be on finalized, right? If so then it probably make sense to not set the commitment to finalized here either.

File: src/metrics/sol/startSubscription.ts [hunk 2]
```diff
@@ -35,12 +35,19 @@ export const startSubscription = async (context: Context) => {
...
+                    });
+                    const keys = transaction.transaction.message.accountKeys;
+                    // only report reverted tx if they are originated from our aggKey
+                    if (keys[0].toString() === global.solanaCurrentOnChainKey) {
```
Comment (albert-llimos) [inline src/metrics/sol/startSubscription.ts:45]: I assume the `global.solanaCurrentOnChainKey` is bs58 encoded, so taking the SC key and encoding it, right?

File: src/metrics/sol/startSubscription.ts (hunk 1 above)
Comment (marcellorigotti) [reply src/metrics/sol/startSubscription.ts:41]: Yes it is on finalized!

File: src/metrics/sol/startSubscription.ts (hunk 2 above)
Comment (marcellorigotti) [reply src/metrics/sol/startSubscription.ts:45]: exactly, this is how we define it:
```
const onChainKey = (await api.query.solanaBroadcaster.currentOnChainKey()).toJSON();
global.solanaCurrentOnChainKey = base58.encode(hexToU8a(onChainKey));
```

**Review by albert-llimos — APPROVED** [review]

**Pull request merged**